language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/traces/LogParser.java

```diff
@@ -90,19 +90,19 @@ private static String removePayload(String header, String payload) {
      * @param logLine String
      * @return log Trace
      */
-    static Trace fromString(String logLine) {
-        Trace log = new Trace();
-        log.setId(getId(logLine));
-        log.setDirection(getDirection(logLine));
+    static Message fromString(String logLine) {
+        Message message = new Message();
```

Comment:
Shall we make `Message` immutable? Then we can remove all setters `setId , setDirection...etc`.
```
return new Message(getId(logLine), 
    getDirection(logLine), 
    removePayload(header, payload), 
    headerType, getContentType(logLine), 
    getHttpMethod(logLine), 
    getPath(logLine), payload);
```